tests/test_commit.py

```diff
@@ -326,3 +330,141 @@ def test_commit_schema_violation(cli_runner, data_working_copy):
+            )
+            sess.execute(
+                f"""INSERT INTO {new_table} (test_id, field1, field2)
+                VALUES
+                    (1, 'value1a', 'value1b'),
+                    (2, 'value2a', 'value2b'),
+                    (3, 'value3a', 'value3b'),
+                    (4, 'value4a', 'value4b'),
+                    (5, 'value5a', 'value5b');"""
+            )
+
+        r = cli_runner.invoke(["add-dataset", new_table, "-m", message, "-o", "text"])
+
+        assert r.exit_code == 0, r
+
+        diff = {
+                new_table: {
+                    "meta": {
+                        "inserts": 1
+                    },
+                    "feature": {
+                        "inserts": 5
+                    }
+                }
+            }
+
+        flat_diff = ""
+        for table, table_diff in diff.items():
+            flat_diff += f"  {table}:\n"
+            for section, section_diff in table_diff.items():
+                for op, count in section_diff.items():
+                    flat_diff += f"    {section}:\n"
+                    flat_diff += f"      {count} {op}\n"
+
+        date = datetime.now().strftime('%a %b %d %H:%M:%S %Y %z+1300\n')
+        expected_output = f"[main {str(repo.head.target)[:7]}] {message}\n{flat_diff}  Date: {date}"
+
+        assert r.stdout == expected_output
+
+
+def test_commit_table_nonexistent(cli_runner, data_working_copy):
+    new_table = "test_table"
+    message = "test commit"
+    with data_working_copy("points") as (path, wc):
+        repo = KartRepo(path)
+        with repo.working_copy.tabular.session() as sess:
+            sess.execute(
+                f"""CREATE TABLE IF NOT EXISTS {new_table} (test_id int primary key, field1 text, field2 text);"""
+            )
+
+        try:
+            cli_runner.invoke(["add-dataset", "wrong_test_table", "-m", message, "-o", "text"])
+        except NotFound as e:
+            assert str(e) == f"Table wrong_test_table is either not found or already tracked by kart.\n\nTry running 'kart status --list-untracked-tables'\n"
+            assert e.exit_code == NO_CHANGES
+
+
+def test_commit_table_twice(cli_runner, data_working_copy):
+    new_table = "test_table"
+    message1 = "test commit1"
+    message2 = "test commit2"
+
+    with data_working_copy("points") as (path, wc):
+        repo = KartRepo(path)
+        with repo.working_copy.tabular.session() as sess:
+            sess.execute(
+                f"""CREATE TABLE IF NOT EXISTS {new_table} (test_id int primary key, field1 text, field2 text);"""
+            )
+
+        try:
+            cli_runner.invoke(["add-dataset", "test_table", "-m", message1, "-o", "text"])
+            cli_runner.invoke(["add-dataset", "test_table", "-m", message2, "-o", "text"])
+        except NotFound as e:
+            assert str(e) == f"Table wrong_test_table is either not found or already tracked by kart.\n\nTry running 'kart status --list-untracked-tables'\n"
```

Comment:
could we differentiate the two cases (table doesn't exist, table already tracked by kart)? I think I'd find this error message a bit confusing as a user